### PR TITLE
Enable `check_path_exists` to work with wildcards.

### LINKED ISSUE #430: FileNotFoundError: ...EUMETSAT/SEVIRI_RSS/zarr/v2/eumetsat_zarr_* does not exist!
When running `prepare_ml_data.py` just now on `leonardo`....

```
2021-11-17 17:27:41,169 DEBUG processID=1551253 Creating satellite DataSource object. | /home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py#L103
DEBUG:nowcasting_dataset.manager:Creating satellite DataSource object.
2021-11-17 17:27:41,170 ERROR processID=1551253 Exception whilst instantiating satellite! | /home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py#L119
Traceback (most recent call last):
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py", line 117, in initialise_data_sources
    data_source = data_source_class(**config_for_data_source)
  File "<string>", line 8, in __init__
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 30, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 373, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 345, in __post_init__
    super().__post_init__()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 51, in __post_init__
    self.check_input_paths_exist()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 378, in check_input_paths_exist
    nd_fs_utils.check_path_exists(self.zarr_path)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/filesystem/utils.py", line 98, in check_path_exists
    raise FileNotFoundError(f"{path} does not exist!")
FileNotFoundError: /mnt/storage_ssd_8tb/data/ocf/solar_pv_nowcasting/nowcasting_dataset_pipeline/satellite/EUMETSAT/SEVIRI_RSS/zarr/v2/eumetsat_zarr_* does not exist!
ERROR:nowcasting_dataset.manager:Exception whilst instantiating satellite!
Traceback (most recent call last):
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py", line 117, in initialise_data_sources
    data_source = data_source_class(**config_for_data_source)
  File "<string>", line 8, in __init__
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 30, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 373, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 345, in __post_init__
    super().__post_init__()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 51, in __post_init__
    self.check_input_paths_exist()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 378, in check_input_paths_exist
    nd_fs_utils.check_path_exists(self.zarr_path)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/filesystem/utils.py", line 98, in check_path_exists
    raise FileNotFoundError(f"{path} does not exist!")
FileNotFoundError: /mnt/storage_ssd_8tb/data/ocf/solar_pv_nowcasting/nowcasting_dataset_pipeline/satellite/EUMETSAT/SEVIRI_RSS/zarr/v2/eumetsat_zarr_* does not exist!
Traceback (most recent call last):
  File "/home/jack/dev/ocf/nowcasting_dataset/scripts/prepare_ml_data.py", line 79, in <module>
    main()
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/utils.py", line 155, in inner_func
    return func(*args, **kwargs)
  File "/home/jack/dev/ocf/nowcasting_dataset/scripts/prepare_ml_data.py", line 67, in main
    manager.initialise_data_sources(names_of_selected_data_sources=data_source)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py", line 117, in initialise_data_sources
    data_source = data_source_class(**config_for_data_source)
  File "<string>", line 8, in __init__
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 30, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 373, in __post_init__
    super().__post_init__(image_size_pixels, meters_per_pixel)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 345, in __post_init__
    super().__post_init__()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 51, in __post_init__
    self.check_input_paths_exist()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 378, in check_input_paths_exist
    nd_fs_utils.check_path_exists(self.zarr_path)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/filesystem/utils.py", line 98, in check_path_exists
    raise FileNotFoundError(f"{path} does not exist!")
FileNotFoundError: /mnt/storage_ssd_8tb/data/ocf/solar_pv_nowcasting/nowcasting_dataset_pipeline/satellite/EUMETSAT/SEVIRI_RSS/zarr/v2/eumetsat_zarr_* does not exist!
```

### PR DESCRIPTION
**Pull request by JackKelly**
# Pull Request

## Description

Fixes #430 

## How Has This Been Tested?

- [ ] No
- [x] Yes, unit tests pass
- [x] `prepare_ml_data.py` runs (but crashes due to another issue)

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/filesystem/utils.py [hunk 1]
```diff
@@ -92,10 +92,17 @@ def delete_all_files_in_temp_path(path: Union[Path, str], delete_dirs: bool = Fa
...
+    `path` can include wildcards.
+    """
     filesystem = get_filesystem(path)
     if not filesystem.exists(path):
-        raise FileNotFoundError(f"{path} does not exist!")
+        # Now try using `glob`.  Maybe `path` includes a wildcard?
+        # Try `exists` before `glob` because `glob` might be slower.
+        files = filesystem.glob(path)
+        if len(files) == 0:
+            raise FileNotFoundError(f"{path} does not exist!")
```
Comment (peterdudfield) [inline nowcasting_dataset/filesystem/utils.py:105]: can you add test to prove it works? 

Or I can

Comment (JackKelly) [reply nowcasting_dataset/filesystem/utils.py:105]: ooh, yeah, that'd be great if you could add a test and merge it into this branch?  Currently chasing another bug...

Comment (peterdudfield) [reply nowcasting_dataset/filesystem/utils.py:105]: Ill do that

**Review by jacobbieker — APPROVED** [review]
Looks good! Thanks for the catch!

File: nowcasting_dataset/filesystem/utils.py (hunk 1 above)
Comment (JackKelly) [reply nowcasting_dataset/filesystem/utils.py:105]: Nice!  Thank you!

**Pull request merged**